Create common site pages

zglueck opened the pull request:
This pull request closes #24 by creating the home, about, and contact pages.

The home page is a partial html template injected in the main block template. The content markdown file doesn't provide any content in this case, but the front matter drives logic in the `list.html` template which injects the partial. To support the tool tips, a short script is added to the main page.

The about and contact pages are straight forward markdown context injected into templates. There are some image tags that we may want to refactor to shortcodes in the future.

A 404 html file was added, and according to Hugo documentation, will work with our gh-pages hosting.

There was an error in the navbar that is fixed. The longer commit history is the result of this branch not merging `develop` at initiation. This is similar to what we saw with the previous pull request.

If there are any suggestions for additional common pages, let me know, we can work them in! Any comments to the layout or any errors you see in your own environments are greatly appreciated.

zglueck committed:
Remove unnecessary page variable checks

zglueck committed:
Add travis configuration
- See #22 for epic context
- See #12 for complete issue

zglueck committed:
Add homepage
- See #24
- Add supporting images
- Turn on tooltips on the home page
- Add the main page as a partial

zglueck committed:
Add contact page
- See #24

zglueck committed:
Add about page
- See #24

zglueck committed:
Add padding to images

zglueck committed:
Add 404 page

zglueck committed:
Fix branding

zglueck committed:
Fix responsive navbar dropdown

zglueck committed:
Merge develop

emxsys commented on content/about.md:
```diff
@@ -0,0 +1,26 @@
+
+WorldWind is a free, open source API for a virtual globe.  Written in Java so as to be cross-platform and easy-to-use, WorldWind allows developers to quickly and easily create interactive visualizations of 3D globe, map and geographical information.
+
+<img align="left" src="/img/800px-worldwind_screen.jpg" width="30%" hspace="10" vspace="10"> WorldWindOrganizations across the world use WorldWind to monitor weather patterns, visualize cities and terrain, track the movement of planes, vehicles and ships, analyze geospatial data, and educate people about the Earth.
```
I think "WorldWindOrganizations" should be "Organizations".

zglueck replied:
Thanks for the catch @emxsys!

emxsys commented on content/about.md:
```diff
@@ -0,0 +1,26 @@
+
+Curated by NASA, work on WorldWind began in 2002, released under the NASA Open Source Agreement (NOSA).  The project was ported to Java in 2006, and in November 2009 was awarded NASA Software of the Year.  Since then it has developed a global community of users, from government, education, the non-profit sector, as well as private industry.
+
+WorldWind is different from a 3D globe like Google Earth because it is not a completed application targeted at end users.  Instead, it is an SDK (software development kit) that software engineers can use to build their own applications.  WorldWind provides a geographic rendering engine for powering a wide range of projects, from satellite tracking systems to flight simulators.
```
Minor suggestion to remove negative context: Change "WorldWind is different from a 3D globe like Google Earth because it is not a completed application targeted at end users." to "WorldWind is different from a 3D globe like Google Earth because it is not an application targeted at specific end users."

zglueck replied:
I've tried simplifying it to remove the context with 86ef890262. Any additional remarks are welcomed!

emxsys commented on layouts/partials/home.html:
```diff
@@ -0,0 +1,69 @@
+            </div>
+            <div class="col-md-4">
+                <h3 class="text-center">Build what you want</h3>
+                <p>WorldWind is different from a 3D globe like Google Earth because it is not a completed application targeted at end users. Instead, it is an SDK (software development kit) that software engineers can use to build their own applications. WorldWind provides a geographic rendering engine for powering a wide range of projects, from satellite tracking systems to flight simulators.</p ...
```
Minor suggestion to remove negative context: Change "WorldWind is different from a 3D globe like Google Earth because it is not a completed application targeted at end users." to "WorldWind is different from a 3D globe like Google Earth because it is not an application targeted at specific end users."

zglueck committed:
Update wording
- See discussion with @emxsys in #41

emxsys commented on content/about.md:
```diff
@@ -11,15 +11,15 @@ listdescription: ""
+WorldWind is a free, open source API for visualizing and hosting geospatial data. Our client libraries, available for Android devices, the browser, or the desktop, provide developers a quick and easy utility for visualizations of 3D globe, map and geographical information. Our server kit extends the functionality of GeoServer to provide an end to end geospatial solution.
 
-<img align="left" src="/img/800px-worldwind_screen.jpg" width="30%" hspace="10" vspace="10"> WorldWindOrganizations across the world use WorldWind to monitor weather patterns, visualize cities and terrain, track the movement of planes, vehicles and ships, analyze geospatial data, and educate people about the Earth.
+<img align="left" src="/img/800px-worldwind_screen.jpg" width="30%" hspace="10" vspace="10"> WorldWind Organizations across the world use WorldWind to monitor weather patterns, visualize cities and terrain, track the movement of planes, vehicles and ships, analyze geospatial data, and educate people about the Earth.
```
I think WorldWind should be deleted from "WorldWind Organizations".

pdavidc replied:
Bruce is right. "WorldWind Organizations" looks like a copy-and-paste error.

emxsys reviewed (commented):
I like the new treatment, especially the opening paragraph.

pdavidc commented on content/about.md:
```diff
@@ -0,0 +1,26 @@
+
+WorldWind is a free, open source API for visualizing and hosting geospatial data. Our client libraries, available for Android devices, the browser, or the desktop, provide developers a quick and easy utility for visualizations of 3D globe, map and geographical information. Our server kit extends the functionality of GeoServer to provide an end to end geospatial solution.
+
+<img align="left" src="/img/800px-worldwind_screen.jpg" width="30%" hspace="10" vspace="10"> WorldWind Organizations across the world use WorldWind to monitor weather patterns, visualize cities and terrain, track the movement of planes, vehicles and ships, analyze geospatial data, and educate people about the Earth.
```
Let's use the actual size for these images. They don't look good when scaled.

zglueck replied:
I generated new size appropriate screen shots from WebWW and WWA. I added the 'img-responsive' class, which will shrink the images as the screen size reduces, but since the images are starting at more reasonable resolutions, I don't think the distortion is too bad.

pdavidc commented on content/contact.md:
```diff
@@ -0,0 +1,30 @@
+---
+
+### Contact
+If you would like to reach out to the NASA WorldWind team, please try the following:
```
Let's drop this text. I think it's superfluous. My impression of good modern web design is that sites omit text that is redundant. For example, we'd omit text that's made redundant by the page's context, the page headings, and the page layout.

pdavidc commented on content/contact.md:
```diff
@@ -0,0 +1,30 @@
+If you would like to reach out to the NASA WorldWind team, please try the following:
+
+---
+#### NASA Program Management
```
I think Patrick wanted this to read NASA Administrative Contact, like it did on the current site at https://worldwind.arc.nasa.gov.

pdavidc commented on content/contact.md:
```diff
@@ -0,0 +1,30 @@
+---
+#### NASA Program Management
+
+The WorldWind project is managed from the NASA Ames center in Sunnyvale, CA. The government lead and point of contact:
```
The location is "NASA's Ames Research Center at Moffett Field, CA."
https://www.nasa.gov/ames/visitorcenter.html

pdavidc commented on content/contact.md:
```diff
@@ -0,0 +1,30 @@
+---
+
+#### Issue Reports and Source Code
+WorldWind uses [GitHub](https://github.com/NASAWorldWind/) for source and issue management. Please head on over to check out the latest code or report issues or requests.
```
Let's keep our text simple and lean, and drop the second sentence.

pdavidc commented on content/contact.md:
```diff
@@ -0,0 +1,30 @@
+---
+
+#### Development Community
+The [WorldWind Forum](https://forum.worldwindcentral.com/) is a great resource to learn more about the WorldWind projects. Many users are quick to help and share code snippets.
```
In this case the additional text is really helpful. I think it provides a better impression of what kind of help can be found on the forum.

pdavidc replied:
You could mention that the NASA and ESA teams regularly visit the forum to answer questions and help with issues.

pdavidc commented on layouts/404.html:
```diff
@@ -0,0 +1,11 @@
+{{ define "head-title" }}
```
I don't see this clever 404 page. I'm using `hugo serve` to view the site.

zglueck replied:
I'm not sure how to get the `hugo serve` functionality to show the 404 page. I was following [their directions](https://gohugo.io/templates/404/) for how it was supposed to work in GitHub pages.

pdavidc replied:
Thanks for clarifying. Based on the directions you linked to, we may need to configure Apache to display this page. We can certainly deploy this, then adjust as necessary.

pdavidc commented on layouts/_default/list.html:
```diff
@@ -12,9 +12,21 @@
     {{ end }}
 {{ end }}
 
+{{ define "head-block" }}
```
This effect is great, but it has a design issue and an architectural issue:
- Design: this does not work on mobile devices. I think we drop it for now and revisit how we want to launch demos.
- Architecture: list.html would become too complex with this addition. If you're not convinced we should drop this change outright, could we inject it through a partial?

zglueck replied:
The tooltips were an attempt to mimic what the current website shows below each button. I've removed for now, if we want to pull that text back in, I will adjust.

pdavidc commented on layouts/partials/home.html:
```diff
@@ -0,0 +1,69 @@
+                </div>
+                <div class="row">
+                    <div class="col-md-8">
+                    <p>WorldWind is a virtual globe technology providing the means to visualize, manipulate and analyze data in a real world virtual representation.</p>
```
The alignment with the two text rows and the buttons is off. Also, as I look at this is a new layout, I think the second row of text may be superfluous. How would it look if we just had the set "Welcome to NASA WorldWind" or "NASA WorldWind", made the text larger, and used a color that stood out better against the background?

I realize this is a straight port from the site currently at https://worldwind.arc.nasa.gov, and may be an existing issue.

pdavidc commented on layouts/partials/home.html:
```diff
@@ -0,0 +1,69 @@
+                <p>Build a cross-platform geospatial desktop application in Java. For lots of examples and documentation see the Developer Guide. Or clone the <a href="https://github.com/NASAWorldWind/WorldWindJava">WorldWindJava GitHub repository</a> and start coding.</p>
+            </div>
+        </div>
+    </div>
```
Is there a way to have some empty space at the bottom of the page?

pdavidc reviewed (changes requested):
Fantastic start to the new site. I have many comments and requested changes that I think will make what you've done even better.

zglueck committed:
Update style and content

Changes are from discussion with @pdavidc and @emxsys on #41.

- Remove second text block from home page header
- Remove tooltips on buttons in home page header
- Remove full page width scaling of header image
- Update images on about page
- Remove extra text in contact page
- Add padding to bottom of the page

zglueck committed:
Adjust jumbotron font

pdavidc commented:
Very nicely done, @zglueck.
